src/state/logging.ts

```diff
@@ -0,0 +1,57 @@
+import * as Sentry from '@sentry/react'
+
+import { AppState } from './types'
+
+/* Utility type to mark all properties of a type as optional */
+type Partial<T> = T extends object
+  ? {
+      [P in keyof T]?: Partial<T[P]>
+    }
+  : T
+
```

Comment:
Call this `DeepPartial`, as `Partial` is already a utility type.
```suggestion
type DeepPartial<T> = T extends object
  ? {
      [P in keyof T]?: DeepPartial<T[P]>
    }
  : T

```

Reply:
Haha, had it like this before actually!